misk-policy/src/main/kotlin/misk/policy/opa/OpaApi.kt

```diff
@@ -0,0 +1,23 @@
+package misk.policy.opa
+
+import okhttp3.ResponseBody
+import retrofit2.Call
+import retrofit2.http.Body
+import retrofit2.http.Headers
+import retrofit2.http.POST
+import retrofit2.http.Path
+
+interface OpaApi {
+  @Headers("Content-Type: application/json")
+  @POST("/v1/data/{documentPath}")
+  fun queryDocument(
+    @Path("documentPath") documentPath: String,
+  ): Call<ResponseBody>
+
+  @Headers("Content-Type: application/json")
+  @POST("/v1/data/{documentPath}")
+  fun queryDocument(
+    @Path("documentPath") documentPath: String,
+    @Body input: String
```

Comment:
out of curiosity, is it possible to change `input` to `@Body input: String?` and then combine these 2 methods?

Reply:
Retrofit @Body parameters must be non-null. I think I tried changing this to `@Body input: String = ""`, but I don't remember if that worked of if OPA had issues with the resulting request. I'll try it again and see if I can collapse them.

Reply:
I went with a single retrofit API and pass in an empty body in the case where no input is required